src/middlewared/middlewared/plugins/nfs.py

```diff
@@ -77,6 +102,72 @@ class Config:
         Bool('managed_nfsd', default=True),
     )
 
+    @private
+    def setup_directories(self):
+        '''
+        We are moving the NFS state directory from /var/lib/nfs to
+        the system dataset: /var/db/system/nfs.
+        When setup_directories is called /var/db/system/nfs is expected to exist.
+
+        If STATEDIR is empty, then this might be an initialization
+        and there might be current info in /var/lib/nfs.
+
+        We always make sure the expected directories are present
+        '''
+        def name_to_id_conversion(usrgrp):
```

Comment:
let's move the `name_to_id_conversion` inner method to its own `@private` method in this class. Using inner methods actually incurs a non-trivial amount of overhead. You can keep the same name if you want. I also see that if `usergrp[0]` fails to be resolved, then we won't try and resolve the group (usergrp[1]). I'm not sure if that's by design but I believe we should make the method a bit more robust and explicit. I also see that the log message is talking about "applying ownership" but it's doing none of the sorts. Looks like it's just resolving the name to a uid/gid 😄 I picture the method to look something like:
```
@private
def name_to_id_converstion(name, name_type='user'):
    if any((not isinstance(name, str), isinstance(name, int))):
        # it's not a string (NoneType, float, w/e) or it's an int
        # so there is nothing to do
        return name

    if name_type = 'user':
        method = 'user.get_builtin_user_id'
    elif name_type == 'group':
        method = 'group.get_builtin_group_id'
    else:
        self.logger.error('Unexpected name_type (%r)', name_type)
        return name
    try:
        return self.middleware.call_sync(method, name)
    except Exception as e:
        if hasattr(e, 'errno') and e.errno == errno.ENOENT:
            self.logger.error('Failed to resolve builtin %s %r', name_type, name)
        else:
             self.logger.error('Unexpected error resolving builtin %s %r, name_type, name, exc_info=True)
        return name